geometric_kernels/spaces/eigenfunctions.py

```diff
@@ -119,6 +119,11 @@ def dim_of_eigenspaces(self) -> B.Numeric:
         """Dimension of eigenspaces. For generic Eigenfunctions, it is an array of ones. For Eigenfunctions with addition theorem, it is the number of eigenfunctions in each level."""
         return B.ones(self.num_eigenfunctions)
 
+    @abc.abstractproperty
+    def num_eigenfunctions_per_level(self) -> B.Numeric:
```

Comment:
TBH I don't understand why this works. For hyperspheres, for example, this function returns a list, and I believe that list is not an instance of `B.Numeric`.
<img width="671" alt="image" src="https://github.com/GPflow/GeometricKernels/assets/638432/fdd7002a-d485-48da-868d-11ba7a87c8cf">

Reply:
Interesting! Maybe we need a stricter mypy config